### PR TITLE
Added volume tween for audio media objects.

### PR DESCRIPTION
**Pull request by aaronpcz**
- Tween dep had been moved to a new npm namespace.
- All old js files using old lib also updated
- Full Scene Store was doing a double update emit, causing us extra update cycles not required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,9 +21,10 @@
   },
   "peerDependencies": {},
   "dependencies": {
+    "@tweenjs/tween.js": "^17.2.0",
     "@vimeo/player": "^1.0.6",
     "async": "^2.3.0",
-    "audio5": "^0.1.10",
```
Comment (aaronpcz) [inline package.json]: The updates have been merged into the npm version of the library.

I'll flip this to the new release version via npm

File: src/js/components/viewer/models/audio-media-object.jsx [hunk 2]
```diff
@@ -200,43 +216,51 @@ var AudioMediaObject = React.createClass({
...
+        if(this.state.transitionToDoneAudioTween) {
+            this.state.transitionToDoneAudioTween.stop();
+        }
+
         // APEP Ensure the Audio5 component cleans up
         if(this.state.player) {
             this.state.player.pause();
             this.state.player.destroy();
             this.state.player.off("timeupdate", this.audioPlayerTimeUpdate);
-            this.state.player = null;
```
Comment (aaronpcz) [inline src/js/components/viewer/models/audio-media-object.jsx]: I removed this for two reasons:

1. State update patterns during unmounting is not valid.
2. This is not a valid way to change state for a react component.

**Comment by aaronpcz** [comment]
The audio5 library while accepting the pull request has not done an npm release.

I propose we can go ahead with reference to my version of the lib and we can update later.

Please give this a test in the browser as well as reviewing the implemented code.

**Comment by aaronpcz** [comment]
Just received word on an npm release for the audio5 lib.

Will switch deps back to the npm one.

File: src/js/components/viewer/models/audio-media-object.jsx [hunk 3]
```diff
@@ -149,23 +164,24 @@ var AudioMediaObject = React.createClass({
...
 
-                this.play();
-
-                //APEP tween volume TODO Turn back on with time to diagnose
+                //APEP tween volume
                 var _ops = self.props.data.mediaObject._ops;
-                // self.state.playingAudioTween = self.tweenAudio(0, volume, _ops.transitionDuration, this);
-                // self.state.playingAudioTween.start();
+                var firstPlayAudioTween = self.tweenAudio(0, volume, _ops.transitionDuration, this);
+                self.setState({firstPlayAudioTween: firstPlayAudioTween});
```
Comment (Stormgun) [inline src/js/components/viewer/models/audio-media-object.jsx]: I have tested this against the scene AP_BEACH_FRONT_SOUNDS, all sounds were playing full volume.
I have reset cache, node modules and dist, that did not change the result might be worth investigating

File: src/js/components/viewer/models/audio-media-object.jsx [hunk 4]
```diff
@@ -41,11 +39,14 @@ var AudioMediaObject = React.createClass({
...
         return new TWEEN.Tween(position)
             .to(target, duration)
-            .onUpdate(function() {
-                if(this.volume){
-                    this.volume(position.vol);
+            .onUpdate(function(obj) {
+                try {
+                    player.volume(obj.vol);
+                } catch (e) {
+                    console.log("ERROR for onUpdate");
```
Comment (Stormgun) [inline src/js/components/viewer/models/audio-media-object.jsx]: I got a certain pattern of steps that cause the audio object to error, and a console error saying that we are setting the state to an unexisting component.
Steps: 
1.Have Media Browser and Scene viewer components open. Select  AP_BEACH_FRONT_SOUNDS from the scene selector
2.Refresh the page and the results are assumed to be automatically selected  AP_BEACH_FRONT_SOUNDS .Your components look like they have the scene loaded.
3. Go to scene list and select the none option, follow up by selecting AP_BEACH_FRONT_SOUNDS again. You should receive errors in the console

**Pull request merged**